### PR TITLE
Initial commit of pyright

### PR DESCRIPTION
**Pull request by fsargent**
https://microsoft.github.io/pyright/#/
https://github.com/microsoft/pyright

Static Type Checker for Python
Pyright is a full-featured, standards-based static type checker for Python. It is designed for high performance and can be used with large Python source bases.

Pyright includes both a [command-line tool](https://microsoft.github.io/pyright/#/command-line) and an [extension for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=ms-pyright.pyright).

### CODE REVIEW

File: linters/pyright/pylint.test.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { customLinterCheckTest, linterCheckTest, TestCallback } from "tests";
+import { TEST_DATA } from "tests/utils";
+
+linterCheckTest({ linterName: "pyright" });
+
+// pyright "info" and "fatal" severities only show up when certain rules are enabled in .pyrightrc.
+// To test that these levels are appropriately parsed, use the following test setup.
+const preCheck: TestCallback = (driver) => {
+  driver.moveFile(path.join(TEST_DATA, ".pyrightrc"), ".pyrightrc");
+};
```
Comment (TylerJang27) [inline linters/pyright/pylint.test.ts:11]: Looks like you forgot to commit this config file from test data.

File: linters/pyright/pylint.test.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import path from "path";
+import { customLinterCheckTest, linterCheckTest, TestCallback } from "tests";
+import { TEST_DATA } from "tests/utils";
+
+linterCheckTest({ linterName: "pyright" });
```
Comment (TylerJang27) [inline linters/pyright/pylint.test.ts:5]: `linter<type>Test` and `customLinter<type>Test` don't always interact super cleanly with snapshot generation. I think we've addressed all the problems, and this should be fine, but if you run into any problems, happy to help. You may need to rerun the test and commit a new snapshot file.

File: linters/pyright/output.json
Comment (TylerJang27) [inline linters/pyright/output.json:1]: I don't think this file is necessary? If you'd like, you can include a portion of it as a comment in the parser file for future reference.

File: linters/pyright/test_data/pyright_v1.1.304_basic.check.shot [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      "level": "LEVEL_HIGH",
+      "line": "1",
+      "linter": "pyright",
+      "message": ""Enum" is unknown import symbol",
+      "targetType": "python",
+    },
+    {
+      "column": "13",
+      "file": "test_data/basic.in.py",
+      "issueUrl": "http://pyright-messages.wikidot.com/messages:",
```
Comment (TylerJang27) [inline linters/pyright/test_data/pyright_v1.1.304_basic.check.shot:20]: Where'd you get the template for issueURL? Looks like these aren't resolving correctly.

File: readme.md [hunk 4]
```diff
@@ -36,47 +36,47 @@ make the discovery, management and integration of new tools as straight-forward
...
+| Kotlin          | [detekt]<sup><a href="#note-detekt">1</a></sup>, [ktlint]                                                |
+| Kubernetes      | [kube-linter]                                                                                            |
+| Lua             | [stylua]                                                                                                 |
+| Markdown        | [markdownlint], [remark-lint]                                                                            |
+| Nix             | [nixpkgs-fmt]                                                                                            |
+| package.json    | [sort-package-json]                                                                                      |
+| PNG             | [oxipng]                                                                                                 |
+| Protobuf        | [buf] (breaking, lint, and format), [clang-format], [clang-tidy]                                         |
+| Python          | [autopep8], [bandit], [black], [flake8], [isort], [mypy], [pylint], [semgrep], [yapf], [ruff], [pyright] |
```
Comment (TylerJang27) [inline readme.md:67]: Please alphebetize :)

**Review by TylerJang27 — COMMENTED** [review]
Thanks for the contribution! Good start, looks like it just needs some cleanup in a few places. Lmk when you're ready for next review or if you need help debugging.

Also, don't worry about the failing `Trunk Check`; we're investigating.

File: linters/pyright/pylint.test.ts (hunk 1 above)
Comment (fsargent) [reply linters/pyright/pylint.test.ts:11]: Ah don't need it actually. Didn't know how to remove this without failing the whole file.

File: linters/pyright/test_data/pyright_v1.1.304_basic.check.shot (hunk 3 above)
Comment (fsargent) [reply linters/pyright/test_data/pyright_v1.1.304_basic.check.shot:20]: Fixed. I did a crappy /pylint/pyright/ replacement and didn't fix this.

**Review by TylerJang27 — COMMENTED** [review]
Looks great! The only remaining piece seems to be fixing the snapshots. When that's passing, I can go ahead and stamp and merge!

**Comment by TylerJang27** [comment]
Hi @fsargent, did you need any additional help getting this finalized?

**Comment by fsargent** [comment]
Will get this fixed up tomorrow.

**Comment by fsargent** [comment]
@TylerJang27 I think this is ready now.

**Pull request merged**